### ISSUE TITLE
`:` mode file input causes segfault with non-json data

### ISSUE THREAD

**Issue by ralight** [jpmens/jo#95]
```
$ echo "jo likes json" > file.txt
$ jo likes=:./file.txt
Segmentation fault (core dumped)
```

Backtrace:
```
#0  0x0000555555557aeb in json_append_member (object=0x55555555e2a0, key=0x7fffffffe1a9 "likes", value=0x0) at json.c:592
#1  0x0000555555556336 in member_to_object (object=0x55555555e2a0, flags=0, key_delim=0 '\000', kv=0x7fffffffe1a9 "likes") at jo.c:459
#2  0x0000555555556413 in append_kv (object_or_array=0x55555555e2a0, flags=0, key_delim=0 '\000', kv=0x7fffffffe1a9 "likes")
    at jo.c:481
#3  0x00005555555568ea in main (argc=2, argv=0x7fffffffde00) at jo.c:673
```

Prior to https://github.com/jpmens/jo/commit/879c1707d542a7740f7c92af0c165284bcffe75b this produced `{"likes":":./file.txt"}` which could be considered "wrong" as well, but may be better than a segfault.

The cause seems to be calls like `json_append_element(op, vnode(value, flags));` not liking the null value being returned from `vnode()` when the json decoding fails.
```

**Comment by jpmens** [comment]
@cylon can you spare a bit of time to look at this, please?

**Comment by jpmens** [comment]
Closed via #96 and #97
